actors/miner/tests/compact_partitions_test.rs

```diff
@@ -0,0 +1,269 @@
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_to_compact_partitions_with_unproven_sectors() {
+    let (mut h, mut rt) = setup();
+
+    // Wait until deadline 0 (the one to which we'll assign the
+    // sector) has elapsed. That'll let us commit, prove, then wait
+    // finality epochs.
+    let state: State = rt.get_state();
+    let deadline_epoch = new_deadline_info(rt.policy(), state.proving_period_start, 0, rt.epoch)
+        .next_not_elapsed()
+        .next_open();
+    rt.set_epoch(deadline_epoch);
+
+    // create 2 sectors in partition 0
+    h.commit_and_prove_sectors(&mut rt, 2, DEFAULT_SECTOR_EXPIRATION, vec![10, 20], true);
+
+    // Wait WPoStProofChallengePeriod epochs so we can compact the sector.
+    let target_epoch = rt.epoch + rt.policy().wpost_dispute_window;
+    h.advance_to_epoch_with_cron(&mut rt, target_epoch);
+
+    let partition_id = 0;
+    let deadline_id = 0;
+
+    let result = h.compact_partitions(&mut rt, deadline_id, bitfield_from_slice(&[partition_id]));
+    expect_abort_contains_message(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to remove partitions from deadline 0: while removing partitions: cannot remove partition 0: has unproven sectors", result);
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_if_deadline_is_equal_to_wpost_period_deadlines() {
+    let (h, mut rt) = setup();
+    let w_post_period_deadlines = rt.policy().wpost_period_deadlines;
+    let result = h.compact_partitions(&mut rt, w_post_period_deadlines, BitField::default());
+    expect_abort_contains_message(
+        ExitCode::USR_ILLEGAL_ARGUMENT,
+        &format!("invalid deadline {w_post_period_deadlines}"),
+        result,
+    );
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_if_deadline_is_open_for_challenging() {
+    let (h, mut rt) = setup();
+    rt.set_epoch(PERIOD_OFFSET);
+    let result = h.compact_partitions(&mut rt, 0, BitField::default());
+    expect_abort(ExitCode::USR_FORBIDDEN, result);
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn fails_if_deadline_is_next_up_to_be_challenged() {
+    let (h, mut rt) = setup();
+    rt.set_epoch(PERIOD_OFFSET);
+    let result = h.compact_partitions(&mut rt, 1, BitField::default());
+    expect_abort(ExitCode::USR_FORBIDDEN, result);
+
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn deadline_after_next_deadline_should_still_be_open_for_compaction() {
+    let (h, mut rt) = setup();
+    rt.set_epoch(PERIOD_OFFSET);
+    h.compact_partitions(&mut rt, 3, BitField::default()).unwrap();
+    check_state_invariants(&rt);
+}
+
+#[test]
+fn deadlines_challenged_last_proving_period_should_still_be_in_dispute_window() {
+    let (h, mut rt) = setup();
+    rt.set_epoch(PERIOD_OFFSET);
+    let last_proving_period = rt.policy().wpost_period_deadlines - 1;
```

Comment:
curr_deadline - 1 % wpost_period_deadlines

Reply:
@ZenGround0 could you please elaborate on this? it would cause a panic given `current_deadline == 0`, no?